toolchain/check/handle_name.cpp

```diff
@@ -136,18 +146,28 @@ auto HandleNameExpression(Context& context, Parse::Node parse_node) -> bool {
 auto HandleQualifiedDeclaration(Context& context, Parse::Node parse_node)
     -> bool {
   auto pop_and_apply_first_child = [&]() {
-    if (context.parse_tree().node_kind(context.node_stack().PeekParseNode()) !=
-        Parse::NodeKind::QualifiedDeclaration) {
-      // First QualifiedDeclaration in a chain.
-      auto [parse_node1, node_id1] =
-          context.node_stack().PopExpressionWithParseNode();
-      context.declaration_name_stack().ApplyExpressionQualifier(
-          parse_node1, context.FollowNameReferences(node_id1));
-      // Add the QualifiedDeclaration so that it can be used for bracketing.
-      context.node_stack().Push(parse_node);
-    } else {
-      // Nothing to do: the QualifiedDeclaration remains as a bracketing node
-      // for later QualifiedDeclarations.
+    Parse::Node parse_node1 = context.node_stack().PeekParseNode();
```

Comment:
With your changes, it looks like this lambda is only called in one place. Does this need to remain a lambda?

Reply:
Thanks, inlined and simplified.